sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/NumberConverter.scala

```diff
@@ -23,6 +23,13 @@ import org.apache.spark.unsafe.types.UTF8String
 
 object NumberConverter {
 
+  /**
+   * The output string has a max length of one char per bit in the intermediate representation plus
+   * one char for the '-' sign.  This corresponds to the representation of `Long.MinValue` with
+   * `toBase` equal to -2.
+   */
+  private final val MAX_OUTPUT_LENGTH = 64 + 1
```

Comment:
I think use 65 is more suitable.

Reply:
Since decoding is done via a Long interpreted as an unsigned long with an external `-` sign, 
64 + 1 or `java.lang.Long.SIZE + 1`  is intuitive

Reply:
+1 for @gerashegalov 's suggestion `java.lang.Long.SIZE + 1`.